### PR TITLE
Simplify function type signatures

### PR DESCRIPTION
**Pull request by Gudahtt**
The type signatures of many public functions were complex and misleading. This was primarily due to the widespread use of the types `TypedData` and `TypedMessage<T>`, which where inappropriate in most of the places they were used.

The `TypedMessage<T>` type is used by the `signTypedData` function when the version is 'V3' or 'V4'. All references to this type outside of V3 and V4 of `signTypedData` have been removed. This also resulted in the `T` generic parameter being removed from many functions.

The `TypedData` type didn't represent the actual type signature of any function, yet it was used to represent the input data for most functions. In most cases we had no specific expectations for the input type, so `unknown` is now used instead. In the case of `V1` of `signTypedData`, the expected type was always `EIP712TypedData[]`. The `TypedData` type has been reduced to just that one case where it was useful and accurate.

### CODE REVIEW

**Comment by Gudahtt** [comment]
The Node.js v16 failure is unrelated (it also fails on `main`). It is fixed here: https://github.com/MetaMask/eth-sig-util/pull/199

File: src/index.ts [hunk 1]
```diff
@@ -6,7 +6,7 @@ import { intToHex, isHexString, stripHexPrefix } from 'ethjs-util';
 
 import { padWithZeroes } from './utils';
 
-export type TypedData = string | EIP712TypedData | EIP712TypedData[];
+export type TypedData = EIP712TypedData[];
```
Comment (rekmarks) [inline src/index.ts:9]: If `TypedData` is only used in / for V1, can we change its name to `V1TypedData` or something similar?

**Review by rekmarks — CHANGES_REQUESTED** [review]
In addition to the inline request, I think we should add docstrings to `TypedData` and `TypedMessage` noting what (i.e. which versions) they're for.

File: src/index.ts (hunk 1 above)
Comment (Gudahtt) [reply src/index.ts:9]: We could :thinking: I had left it as-is to minimize breaking changes, but I guess we're past that mattering now.

Comment (Gudahtt) [reply src/index.ts]: Updated! It's now called `TypedDataV1`

**Comment by Gudahtt** [comment]
Good idea! The latest commit adds docs for both of those types.

**Review by rekmarks — APPROVED** [review]
LGTM!

**Pull request merged**